snapcraft.yaml

```diff
@@ -122,19 +122,20 @@ parts:
     stage:
       - users-and-groups
       - reserved-usernames
-  kbdnames:
+  keyboard-data:
     plugin: nil
     build-packages:
       - console-setup
       - locales
       - xkb-data-i18n
-    override-build: |
-      /usr/share/console-setup/kbdnames-maker /usr/share/console-setup/KeyboardNames.pl > $SNAPCRAFT_PART_INSTALL/kbdnames.txt
+      - python3-attr
+    override-build: PYTHONPATH=$SNAPCRAFT_PROJECT_DIR/ $SNAPCRAFT_PROJECT_DIR/scripts/make-kbd-info.py
```

Comment:
I'm wondering if we want an integration test step which fails the build if the committed version of any file doesn't match the generated version, sort of like what I did with the schema file but applied more generically.
Put differently, if running the build is known to not change any committed file, then I think we could safely ignore steps like doing make-kbd-info.py at snapcraft time.  What do you think?

Reply:
It's an interesting idea. One thing that's a bit different here is that the keyboard data comes from the console-setup -- if the data is different between focal and hirsute, say, then the test you're proposing would fail on one of these systems. This is something that is actually a change in this branch: previously subiquity parsed the pc105.tree file from the live installer environment, but I think this is a good thing as it seems more proof against that file changing (or moving or disappearing) i a future release of Ubuntu and breaking us.

Of course all of this is a bit theoretical as I don't think any of this stuff is very likely to change any time soon.